Symfony/CS/Fixer/Contrib/EregToPregFixer.php

```diff
@@ -0,0 +1,173 @@
+
+                // did we find a match?
+                if (null === $ereg) {
+                    break;
+                }
+
+                // findSequence also returns the tokens, but we're only interested in the indexes, i.e.:
+                // 0 => function name,
+                // 1 => bracket "("
+                // 2 => quoted string passed as 1st parameter
+                $ereg = array_keys($ereg);
+
+                // advance tokenizer cursor
+                $currIndex = $ereg[2];
+
+                // ensure it's a function call (not a method / static call)
+                $prev = $tokens->getPrevMeaningfulToken($ereg[0]);
+                if (null === $prev || $tokens[$prev]->isGivenKind(array(T_OBJECT_OPERATOR, T_DOUBLE_COLON))) {
+                    continue;
+                }
+
+                // ensure the first parameter is just a string (e.g. has nothing appended)
+                $next = $tokens->getNextMeaningfulToken($ereg[2]);
+                if (null === $next || !$tokens[$next]->equalsAny(array(',', ')'))) {
+                    continue;
+                }
+
+                // convert to PCRE
+                $string = substr($tokens[$ereg[2]]->getContent(), 1, -1);
+                $quote = substr($tokens[$ereg[2]]->getContent(), 0, 1);
+                $delim = $this->getBestDelimiter($string);
+                $preg = $delim.addcslashes($string, $delim).$delim.'D'.$map[2];
+
+                // check if the preg is valid
+                if (!$this->checkPreg($preg)) {
+                    continue;
+                }
+
+                // modify function and argument
+                $tokens[$ereg[2]]->setContent($quote.$preg.$quote);
+                $tokens[$ereg[0]]->setContent($map[1]);
+            }
+        }
+
+        return $tokens->generateCode();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getDescription()
+    {
+        return 'Replace deprecated ereg regular expression functions with preg.';
+    }
+
+    /**
+     * Check the validity of a PCRE
+     *
+     * @param string $pattern the regular expression
+     *
+     * @return bool
+     */
+    private function checkPreg($pattern)
+    {
+        return false !== @preg_match($pattern, '');
+    }
+
+    /**
+     * Get the delimiter that would require the least escaping in a regular expression.
+     *
+     * @param string $pattern the regular expression
+     *
+     * @return string the preg delimiter
+     */
+    private function getBestDelimiter($pattern)
+    {
+        // try do find something that's not used
+        $delimiters = array();
+        foreach (self::$delimiters as $k => $d) {
+            if (false === strpos($pattern, $d)) {
+                return $d;
+            }
+
+            $delimiters[$d] = array(substr_count($pattern, $d), $k);
+        }
+
+        // return the least used and/or the first in the map order
+        uasort($delimiters, function ($a, $b) {
```

Comment:
Yes, I know. I just don't like to promote the code that will be copy-pased and used somewhere else, by someone who don't know that two problems with this implementation